TopQuarkAnalysis/TopEventProducers/interface/TopDecaySubset.h

```diff
@@ -43,12 +47,34 @@ class TopDecaySubset : public edm::EDProducer {
  private:
   /// find top quarks in list of input particles
   std::vector<const reco::GenParticle*> findTops(const reco::GenParticleCollection& parts);
+  /// find primal top quarks (top quarks from the hard interaction)
+  /// for Pythia6 this is identical to findDecayingTops
+  std::vector<const reco::GenParticle*> findPrimalTops(const reco::GenParticleCollection& parts);
```

Comment:
Hi,
While the functionality is identical I was told by the TOP PAG conveners not to change run1 code but instead build in switches to enable the new code on demand. I am all in favour in reducing code on the one hand, but on the other I would not replace the code without a unit test verifying its functionality.

Maybe I can add such tests and talk to the conveners. Do you know of any good unit test examples where GenParticleCollections are used? I am not sure how I would construct a mock genparticle collection.

Cheers,
Luke